qa/common/util.sh

```diff
@@ -468,6 +468,16 @@ function kill_servers () {
     done
 }
 
+# Upload a local directory to a GCS path
+function gcs_upload () {
+    local local_path=$1
+    local gcs_path=$2
+    rm -f gcs_upload.gsutil_cp.log
+    until gsutil -m cp -c -L gcs_upload.gsutil_cp.log -r $local_path $gcs_path; do
+        sleep 1
+    done
```

Comment:
No, because if it fails to upload, there is nothing to do but fail the test. In that case, it would be better to keep retrying until it timed out. If it timed out, then we know it is a network issue.

Reply:
What is the timeout? If you are referring to the CI timeout, that will be too long, right?

Reply:
Ok, adding max retry.

Reply:
If fails:
```
+ gcs_upload models/ gs://<hidden>
+ local local_path=models/
+ local gcs_path=gs://<hidden>
+ local retry_left=30
+ local log_file=gcs_upload.gsutil_cp.log
+ rm -f gcs_upload.gsutil_cp.log
+ touch gcs_upload.gsutil_cp.log
+ gsutil -m cp -c -L gcs_upload.gsutil_cp.log -r models/ gs://<hidden>
Copying file://models/dummy [Content-Type=application/octet-stream]...
Error copying file://models/: NotFoundException: 404 The destination bucket gs://<hidden> does not exist or the write to the destination must be restarted
CommandException: 1 file/object could not be transferred.
+ sleep 1
+ (( retry_left-- ))
+ [[ 29 -eq 0 ]]
+ gsutil -m cp -c -L gcs_upload.gsutil_cp.log -r models/ gs://<hidden>
Copying file://models/dummy [Content-Type=application/octet-stream]...
Error copying file://models/: NotFoundException: 404 The destination bucket gs://<hidden> does not exist or the write to the destination must be restarted
CommandException: 1 file/object could not be transferred.
+ sleep 1
+ (( retry_left-- ))
+ [[ 28 -eq 0 ]]
...
+ [[ 0 -eq 0 ]]
+ cat gcs_upload.gsutil_cp.log
... <hidden> ...
+ echo -e '=== Failed upload to GCS bucket'
=== Failed upload to GCS bucket
+ return 121
```

Reply:
If success:
```
+ gcs_upload models/ gs://<hidden>
+ local local_path=models/
+ local gcs_path=gs://<hidden>
+ local retry_left=30
+ local log_file=gcs_upload.gsutil_cp.log
+ rm -f gcs_upload.gsutil_cp.log
+ touch gcs_upload.gsutil_cp.log
+ gsutil -m cp -c -L gcs_upload.gsutil_cp.log -r models/ gs://<hidden>
Copying file://models/dummy [Content-Type=application/octet-stream]...
/ [0/1 files][    0.0 B/    0.0 B]                                              
/ [1/1 files][    0.0 B/    0.0 B]                                              
Operation completed over 1 objects.
...
```